homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,576 @@
+    def find_chlorinator(coordinator, systemid):
+        """Find the correct chlorinator entity."""
+
+        for backyard in coordinator.data:
+            for bow in backyard.get("BOWS"):
+                if bow["Chlorinator"]["systemId"] == systemid:
+                    sensor_data = bow.get("Chlorinator")
+                    break
+
+        return sensor_data
+
+    @staticmethod
+    def find_csad(coordinator, systemid):
+        """Find the correct CSAD entity."""
+
+        for backyard in coordinator.data:
+            for bow in backyard.get("BOWS"):
+                if bow["CSAD"]["systemId"] == systemid:
+                    sensor_data = bow.get("CSAD")
+                    break
+
+        return sensor_data
+
+
+class OmniLogicTemperatureSensor(OmnilogicSensor):
+    """Define an OmniLogic Temperature (Air/Water) Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        backyard: dict,
+        bow: dict,
+        device_class: str,
+        icon: str,
+        unit: str,
+        sensor_data: dict,
+    ):
+        """Initialize the sensor."""
+        backyard_name = backyard["BackyardName"]
+
+        if kind == "water_temperature":
+            system_id = bow["systemId"]
+            name = f"{backyard_name} {bow['Name']} {name}"
+        else:
+            system_id = backyard["systemId"]
+            name = f"{backyard_name} {name}"
+
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            backyard=backyard,
+            bow=bow,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            sensor_data=sensor_data,
+            system_id=system_id,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the temperature sensor."""
+        _LOGGER.debug("Updating state of sensor: %s", self._name)
+
+        sensor_data = None
+
+        if self._kind == "water_temperature":
+            for backyard in self.coordinator.data:
+                for bow in backyard["BOWS"]:
+                    if bow["systemId"] == self._system_id:
+                        sensor_data = bow["waterTemp"]
+                        break
+
+        elif self._kind == "air_temperature":
+            for backyard in self.coordinator.data:
+                if backyard.get("systemId") == self._system_id:
+                    sensor_data = backyard.get("airTemp")
+
+        temp_return = int(sensor_data)
+        temp_state = int(sensor_data)
+        unit_of_measurement = TEMP_FAHRENHEIT
+        if self._backyard["Unit-of-Measurement"] == "Metric":
+            temp_return = round((temp_return - 32) * 5 / 9, 1)
+            unit_of_measurement = TEMP_CELSIUS
+
+        if int(sensor_data) == -1:
+            temp_return = None
+            temp_state = None
+
+        self._attrs["hayward_temperature"] = temp_return
```

Comment:
We shouldn't duplicate the temperature in the state attributes when it's returned as the state.

Reply:
Because of the way HA auto-converts temperatures to the unit set at the system level this doesn't work well for pools. In Canada we use degrees F for our pools, but use degrees C for almost everything else. With the auto conversion we don't get the temperature preference as set in the Hayward system (in my case I prefer F even though my system is C). This allows the user to get the preference set value from the Hayward system and display it if they choose.

Reply:
Ok.